packages/generator/src/options/options.ts

```diff
@@ -139,19 +140,24 @@ export const cliOptions = {
     demandOption: true,
     requiresArg: true
   },
-  serviceMapping: {
+  optionsPerService: {
     alias: 's',
     describe:
-      'Configuration file to ensure consistent names between multiple generation runs with updated / changed metadata files. Will be generated if not existent. By default it will be saved to/read from the input directory as "service-mapping.json".',
+      'Configuration file to ensure consistent names between multiple generation runs with updated / changed metadata files. The configuration allows to set a `directoryName` and `packageName` for every service, identified by the path to the original file. It also makes sure that names do not change between generator runs. If a directory is passed, a `options-per-service.json` file is read/created in this directory.',
     type: 'string',
     coerce: (
       arg: string | undefined,
       options: GeneratorOptions & { config?: string }
-    ) =>
-      resolveRequiredPath(
-        arg ? arg : join(options.inputDir.toString(), 'service-mapping.json'),
-        options
-      )
+    ) => {
```

Comment:
add a named method with some small doc explaining what is done.

Reply:
Also in the OpenApi client you could use this method then.

Reply:
I'll do this after this [PR](https://github.com/SAP/cloud-sdk-js/pull/3431) is merged, since `resolveRequiredPath` was moved to common.